Tools/autotest/common.py

```diff
@@ -742,11 +776,16 @@ def do_set_mode_via_command_long(self):
                 return
         raise AutoTestTimeoutException()
 
-    def mavproxy_do_set_mode_via_command_long(self):
+    def mavproxy_do_set_mode_via_command_long(self, mode, timeout=60):
         base_mode = mavutil.mavlink.MAV_MODE_FLAG_CUSTOM_MODE_ENABLED
-        custom_mode = 4  # hold
+        mode_map = self.mav.mode_mapping()
```

Comment:
This is copy-and-pasted mapping code; it should be a single function call which raises an exception if things go bad.